### PR TITLE
[Concept Entry] AI: Semi-Supervised Learning

### LINKED ISSUE #5970: [Concept Entry] AI: Semi-Supervised Learning
### Reason/inspiration (optional)

We would like a new entry on the `Semi-Supervised Learning` concept under AI. The entry should go in a new file under `docs/content/ai/concepts/machine-learning/terms/semi-supervised-learning/semi-supervised-learning.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by danitellini**
### Description

Topic: AI: Semi-Supervised Learning
File added: `docs/content/ai/machine-learning/semi-supervised-learning/semi-supervised-learning.md`

### Issue Solved

Closes #5970 

### Type of Change

- Adding a new entry
- ~Editing an existing entry (fixing a typo, bug, issues, etc)~
- ~Updating the documentation~

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/ai/concepts/machine-learning/terms/semi-supervised-learning/semi-supervised-learning.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+base_classifier = RandomForestClassifier(n_estimators=100, random_state=42)
+
+# Create the semi-supervised model
+self_training_model = SelfTrainingClassifier(base_classifier)
+
+# Train the model
+self_training_model.fit(X_train, y_train)
+
+# Evaluate the model
+accuracy = self_training_model.score(X_test, y[~unlabeled_mask])
```
Comment (mamtawardhani) [inline content/ai/concepts/machine-learning/terms/semi-supervised-learning/semi-supervised-learning.md:65]: Could you recheck the code once? I'm getting this error:

`Found input variables with inconsistent numbers of samples: [352, 360]`

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @danitellini , thank you for contributing to Codecademy Docs, the entry is nicely written! 😄 

I've suggested one change, could you please review and modify it at your earliest convenience? Thank you! 😃

**Comment by danitellini** [comment]
All set! @mamtawardhani

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @ 😄 
The entry looks good for  a second round of review! 🚀

**Review by Sriparno08 — APPROVED** [review]
Looks good, @danitellini!
